lib/TestBootstrap.lua

```diff
@@ -33,17 +37,25 @@ function TestBootstrap:getModules(root, modules, current)
 	modules = modules or {}
 	current = current or root
 
-	for _, child in ipairs(current:GetChildren()) do
-		if child:IsA("ModuleScript") and child.Name:match("%.spec$") then
+	local function processChild(child, parent)
+		if isSpecScript(child) then
 			local method = require(child)
-			local path = getPath(child, root)
+			local path = getPath(child, parent)
 
 			table.insert(modules, {
 				method = method,
 				path = path
 			})
 		else
-			self:getModules(root, modules, child)
+			self:getModules(parent, modules, child)
+		end
+	end
+
+	if isSpecScript(root) then
+		processChild(root, root.Parent)
+	else
+		for _, child in ipairs(current:GetChildren()) do
+			processChild(child, root)
 		end
 	end
```

Comment:
Because of the extra machinery in `processChild`, we don't need this branch anymore, right?

We should be able to drop this branch and inline `processChild`, which reintroduces this branch.

Reply:
Nope. we do. That is the original loop over folder children